packages/components/src/internal/components/ontology/OntologyTreePanel.tsx

```diff
@@ -18,14 +19,131 @@ export class OntologyPath {
+    customStyles?: any; // Custom styling object that is applied in addition to the base
+    checked?: boolean; // Is check box checked
+    emptyDirectoryText?: string; // Text to show if node is a container, but has no contents
+
+    allowMultiSelect?: boolean; // Flag to enable multi-selection of nodes
+    isEmpty: boolean; // Flag indicating if flag is an empty container
+    isLoading: boolean; // Flag indicating child data is being loaded for node
+
+    showNodeIcon: boolean; // Flag to indicate whether an Icon should be shown for the node
+    useFileIconCls?: boolean; // Class to apply to the Icon
+    renderIcon?: (props: unknown) => React.ReactElement; // Function Component method to render icon element
+    filters: Map<string, PathModel>;
+    onFilterClick: (node: PathModel) => void;
+}
+
+export const OntologyTreeHeader: FC<OntologyTreeHeaderProps> = memo( props => {
+    const {
+        style,
+        onSelect,
+        node,
+        customStyles,
+        emptyDirectoryText,
+        allowMultiSelect,
+        showNodeIcon = true,
+        isEmpty,
+        isLoading,
+        filters,
+        onFilterClick,
+    } = props;
+
+    if (isEmpty) {
+        return <div className="filetree-empty-directory">{emptyDirectoryText}</div>;
+    }
+
+    if (isLoading) {
+        return (
+            <div className="filetree-empty-directory">
+                <LoadingSpinner />
+            </div>
+        );
+    }
+
+    const isDirectory = node?.children !== undefined;
+    const activeColor = node?.active && !allowMultiSelect ? 'lk-text-theme-dark filetree-node-active' : undefined; // $brand-primary and $gray-light
+
+    // Do not always want to toggle directories when clicking a check box
+    // const checkClick = (evt): void => {
+    //     evt.stopPropagation();
+    // };
+
+    return (
+        <span
+            className={
+                'filetree-checkbox-container' +
+                (isDirectory ? '' : ' filetree-leaf-node') +
+                (node.active ? ' active' : '')
+            }
+        >
+            <div style={style.base} onClick={onSelect}>
+                <div className={activeColor}>
+                    <div
+                        className="filetree-resource-row"
+                        style={node.selected ? { ...style.title, ...customStyles.header.title } : style.title}
+                        title={node.name}
+                    >
+                        <div
+                            className={classNames({
+                                'filetree-file-name': !isDirectory,
+                                'filetree-directory-name': isDirectory,
+                            })}
+                        >
+                            {node.name}
+                        </div>
+                    </div>
+                </div>
+            </div>
+            {showNodeIcon && <FilterIcon node={node} filters={filters} onClick={onFilterClick} />}
```

Comment:
hmmm...this line is really the only difference from the main `Header` component. Could we do something like update `Header` so that it does a `{this.props.children}` after this div and then have this do (note that this is untested):
```
return (
    <Header {...props}>
        {showNodeIcon && <FilterIcon node={node} filters={filters} onClick={onFilterClick} />}
    </Header>
)
```